docs/proposals/federated-ruler.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Ruler Tenant Federation"
+linkTitle: "Ruler Tenant Federation"
+weight: 1
+slug: "Ruler Tenant Federation
+---
+
+- Author: [Rees Dooley](https://github.com/rdooley)
+- Date: November 2021
+- Status: Accepted
+
+## Overview
+
+This document aims to describe how to implement the ability to allow rules to cover data from more than a single Cortex tenant, here after refered to as federated rules. Since currently rules are owned by, query data from and save resulting series in the same tenant, this document aims to provide clear delineation of who owns a federated rule, what tenants the federated rule queries data from and where the federated rule saves resulting series.
+
+A federated rule is any rule which contains the `src_tenants` field.
+
+## Reasoning
+
+There are two primary use cases for allowing federated rules which query data from multiple tenants; administration of cortex and composite tenants.
```

Comment:
@rdooley have you considered putting the `src_tenants` on a rule group level instead of rule level?

On one hand, 
* having it on group level reduces repetition between rules. I'd imagine some of the rules will likely read from or write to the same tenants, so it will make them less prone to bugs (and bugs like that can cause gaps in the rules data)
* currently in prometheus rules in a group are evaluated sequentially. So rules listed earlier in a group can be used by the rules later in the group. If one of the earlier rules has a `dest_tenant`, then this invariant will be violated. This will slightly change behaviour.

On the other hand,
* having it on rule level allows for more flexibility. Otherwise, federated and non-federated rules cannot be mixed in the same group. 
* it requires less changes to existing rule groups. Existing rule groups that users want to turn into federated rules can be amended simply by adding a line to a rule. In the case of group-level tenants, rules will likely need to be split up into multiple federated & non-federated groups.

I'm curious as to why you opted for having them per rule.

Reply:
Opted for making `src_tenants` a rule instead of group based field just for the most flexibility.
Since `dest_tenant` has been removed from this proposal the invariant of earlier rules' data being available for later rules still holds.

Reply:
I can see the case for having `src_tenants` at the rule level and the group level, and I had a weakly held opinion prioritizing flexibility.

Reply:
What about allowing `src_tenants` on either rule or rule group? `src_tenants` on a rule would by default inherit the value from the rule group, but can be overwritten per rule.

Reply:
Since my opinion is weakly held, what if we just go with on the rule group instead of per rule as that seems the preference from Dimitar and Andrea?

Reply:
That's fine with me. This was can always be expanded later to per-rule as well if/when needed.

Reply:
I'll get that updated today and also remove the composite tenant bits to scope this down further. Composite tenant support can be worked on separately later if we need it.